back/dora/orientations/tests/test_close_expired_orientations.py

```diff
@@ -0,0 +1,184 @@
+from unittest.mock import patch
+
+from django.core import mail
+from django.core.management import call_command
+from django.test import TransactionTestCase, override_settings
+from django.utils import timezone
+from freezegun import freeze_time
+
+from dora.core.test_utils import make_orientation
+from dora.orientations.models import OrientationStatus
+
+
+@override_settings(ORIENTATION_EXPIRATION_PERIOD_DAYS=30)
+class CloseExpiredOrientationsTestCase(TransactionTestCase):
+    def setUp(self):
+        self.beneficiary_email = "test@email.com"
+        self.starting_date = "2022-01-28"
+
+        with freeze_time(self.starting_date):
+            self.orientation_1_start_date = timezone.now() - timedelta(days=31)
+            self.expired_orientation_1 = make_orientation(
+                creation_date=self.orientation_1_start_date,
+                processing_date=None,
+                beneficiary_email=self.beneficiary_email,
+            )
+
+            self.orientation_2_start_date = timezone.now() - timedelta(days=41)
+
+            self.expired_orientation_2 = make_orientation(
+                creation_date=timezone.now() - timedelta(days=51),
+                processing_date=self.orientation_2_start_date,
+            )
+
+    @staticmethod
+    def call_command():
+        call_command("close_expired_orientations", stdout=StringIO())
+
+    @patch(
+        "dora.orientations.management.commands.close_expired_orientations.send_orientation_expiration_emails"
+    )
+    def test_should_close_expired_orientations_and_send_emails(self, mock_send_emails):
+        with freeze_time("2022-02-01"):
+            expected_processing_time = timezone.now()
+
+            with self.assertNumQueries(7):
+                self.call_command()
+
+            self.expired_orientation_1.refresh_from_db()
+            self.assertEqual(
+                self.expired_orientation_1.status, OrientationStatus.EXPIRED
+            )
+            self.assertEqual(
+                self.expired_orientation_1.processing_date, expected_processing_time
+            )
+
+            self.expired_orientation_2.refresh_from_db()
+            self.assertEqual(
+                self.expired_orientation_2.status, OrientationStatus.EXPIRED
+            )
+            self.assertEqual(
+                self.expired_orientation_2.processing_date, expected_processing_time
+            )
+
+        self.assertEqual(mock_send_emails.call_count, 2)
+        self.assertEqual(
+            mock_send_emails.call_args_list[0][0][0], self.expired_orientation_1
+        )
+        self.assertEqual(
+            mock_send_emails.call_args_list[0][0][1],
+            self.orientation_1_start_date,
+        )
+        self.assertEqual(
+            mock_send_emails.call_args_list[1][0][0], self.expired_orientation_2
+        )
+        self.assertEqual(
+            mock_send_emails.call_args_list[1][0][1],
+            self.orientation_2_start_date,
+        )
```

Comment:
Simplifiable en ceci :
```suggestion
        self.assertEqual(
            mock_send_emails.call_args_list, [call(self.expired_orientation_1, self.orientation_1_start_date), call(self.expired_orientation_2, self.orientation_2_start_date)]
        )
```